specs/core/0_beacon-chain.md

```diff
@@ -1830,7 +1834,9 @@ def update_justification_and_finalization(state: BeaconState) -> None:
 
     # Rotate justified epochs
     state.previous_justified_epoch = state.justified_epoch
+    state.previous_justified_root = state.justified_root
     state.justified_epoch = new_justified_epoch
+    state.justified_root = get_block_root(state, get_epoch_start_slot(new_justified_epoch))
```

Comment:
won't this fail if `state.justified_epoch` is more than 8k slots old and no new justification happens?

Reply:
Right, should conditionally overwrite